src/main/scala/subsystem/HasTiles.scala

```diff
@@ -122,12 +123,7 @@ trait HasTileInterruptSources
       outputRequiresInput = false,
       inputRequiresOutput = false))
   }
-}
-
-/** Source of Non-maskable Interrupt (NMI) input bundle to each tile.
-  */
-trait HasTileNMISources extends InstantiatesTiles { this: BaseSubsystem =>
-  /** NMI input bundle */
+  /** Source of Non-maskable Interrupt (NMI) input bundle to each tile. */
   val tileNMINode = BundleBridgeEphemeralNode[NMI]()
   val tileNMIIONodes: Seq[BundleBridgeSource[NMI]] = {
     Seq.fill(tiles.size) {
```

Comment:
I think we need to make some kind of `NexusSourceNode` that accepts any number of left-hand-side connections so that you don't need this reference to `tiles.size` here. The `miepNode` above is playing a similar trick. Anyways, fine as-is and I will look at handling all three of these similarly in a future PR.